csrc/dispatch.cpp

```diff
@@ -52,6 +52,10 @@ void Val::dispatch(T handler, Val* val) {
         ptr(handler)->handle(val->as<Int>());
         return;
       }
+      if (!std::holds_alternative<PrimDataType>(val->getDataType()->type)) {
+        ptr(handler)->handleUntypedVal(val);
```

Comment:
Wonder why this is called "untyped". Actually, what "untyped" means?

Reply:
"untyped" means it is not an `Scalar<SomeType>` , it is just a plain `Val` without much detailed information.

Reply:
Is it at the moment always an array?

Reply:
Yes, but array of 3 elements and array of 3 elements are different types. And also we can have array of array. So it is actually infinite amount of types.

Reply:
yeah, sure

Reply:
changed